lib/lz4.c

```diff
@@ -1281,10 +1280,18 @@ LZ4_stream_t* LZ4_createStream(void)
     LZ4_STATIC_ASSERT(LZ4_STREAMSIZE >= sizeof(LZ4_stream_t_internal));    /* A compilation error here means LZ4_STREAMSIZE is not large enough */
     DEBUGLOG(4, "LZ4_createStream %p", lz4s);
     if (lz4s == NULL) return NULL;
-    LZ4_resetStream(lz4s);
+    LZ4_initStream(lz4s, sizeof(*lz4s));
     return lz4s;
 }
 
+LZ4_stream_t* LZ4_initStream (void* buffer, size_t size)
```

Comment:
This buffer needs to be aligned, right? We should either return `NULL`, or align into the buffer, and state that the returned pointer might not be at the same address.

Reply:
Yeah, I was hesitating between the 2 options.

This is equivalent to `ZSTD_*_initStatic()`,
and I'm vaguely remembering someone asking me if the return pointer had effectively the same value as the source pointer, because he wanted to use this property to only store one pointer for the later `free()`.

So if it's a natural inclination, then it's better to spell the alignment restriction explicitly, and refuse any input that does not respect it, by returning `NULL`.

Reply:
I think thats the better option